src/libs/PureEpochs.sol

```diff
@@ -27,4 +27,8 @@ library PureEpochs {
     function timeRemainingInCurrentEpoch() internal view returns (uint40) {
         return STARTING_TIMESTAMP + (currentEpoch() * EPOCH_PERIOD) - uint40(block.timestamp);
     }
+
+    function clockMode() internal pure returns (string memory) {
+        return "mode=epoch&epochUnderlyingSource=blockTimestamp&epochStartingTimestamp=1663224162&epochPeriod=1296000&";
```

Comment:
Shouldn't we use `from` instead of `epochStartingTimestamp` to be compliant with EIP-6372?
https://eips.ethereum.org/EIPS/eip-6372

Reply:
No because that only applies to block number mode. `from` is related to the starting clock value. If we cared to use it at all, it would be `1`, but the implied custom formula for "epoch" can only start at 1.